examples/mxnet_imagenet_resnet50.py

```diff
@@ -0,0 +1,287 @@
+# PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import argparse, time, logging, os
+import numpy as np
+import mxnet as mx
+import math
+from mxnet import gluon, nd
+from mxnet import autograd as ag
+from mxnet.lr_scheduler import MultiFactorScheduler
+from mxnet.io import DataBatch, DataIter
+from mxnet.util import makedirs
+from mxnet.gluon.model_zoo.vision import get_model
+
+import horovod.mxnet as hvd
+
+# Initialize Horovod
+hvd.init()
+
+# CLI
+parser = argparse.ArgumentParser(description='Train a model for image classification.')
+parser.add_argument('--benchmark', type=int, default='0',
+                    help='if 0 then feed the network with synthetic data.')
+parser.add_argument('--data-nthreads', type=int, default=4,
+                    help='number of threads for data decoding')
+parser.add_argument('--rec-train', type=str, default='',
+                    help='the training data')
+parser.add_argument('--rec-train-idx', type=str, default='',
+                    help='the index of training data')
+parser.add_argument('--rec-val', type=str, default='',
+                    help='the validation data')
+parser.add_argument('--rec-val-idx', type=str, default='',
+                    help='the index of validation data')
+parser.add_argument('--batch-size', type=int, default=128,
+                    help='training batch size per device (CPU/GPU).')
+parser.add_argument('--dtype', type=str, default='float32',
+                    help='data type for training, default is float32')
+parser.add_argument('--gpus', type=str, default='0',
+                    help='number of gpus to use.')
+parser.add_argument('--num-epochs', type=int, default=90,
+                    help='number of training epochs.')
+parser.add_argument('--lr', type=float, default=0.1,
+                    help='learning rate, default is 0.1.')
+parser.add_argument('--momentum', type=float, default=0.9,
+                    help='momentum value for optimizer, default is 0.9.')
+parser.add_argument('--wd', type=float, default=0.0001,
+                    help='weight decay rate, default is 0.0001.')
+parser.add_argument('--lr-decay', type=float, default=0.1,
+                    help='decay rate of learning rate, default is 0.1.')
+parser.add_argument('--lr-decay-epoch', type=str, default='40,60',
+                    help='epoches at which learning rate decays, default is 40,60.')
+parser.add_argument('--warmup-lr', type=float, default=0.001,
+                    help='starting warmup learning rate, default is 0.001')
+parser.add_argument('--warmup-epochs', type=int, default=5,
+                    help='number of warmup epochs, default is 5.')
+parser.add_argument('--model', type=str, required=True,
```

Comment:
Should this default to `resnet50` due to name of the python script?  Where to get this `vision_model`?

Reply:
updated